src/Type/Php/IsAFunctionTypeSpecifyingHelper.php

```diff
@@ -25,16 +25,22 @@ public function determineType(
 		bool $allowSameClass,
 	): Type
 	{
-		$objectOrClassTypeClassName = $this->determineClassNameFromObjectOrClassType($objectOrClassType, $allowString);
+		$objectOrClassTypeClassNames = $objectOrClassType->getObjectClassNames();
+		if ($allowString) {
+			foreach ($objectOrClassType->getConstantStrings() as $constantString) {
+				$objectOrClassTypeClassNames[] = $constantString->getValue();
+			}
+			$objectOrClassTypeClassNames = array_unique($objectOrClassTypeClassNames);
+		}
 
 		return TypeTraverser::map(
 			$classType,
-			static function (Type $type, callable $traverse) use ($objectOrClassTypeClassName, $allowString, $allowSameClass): Type {
+			static function (Type $type, callable $traverse) use ($objectOrClassTypeClassNames, $allowString, $allowSameClass): Type {
 				if ($type instanceof UnionType || $type instanceof IntersectionType) {
 					return $traverse($type);
 				}
 				if ($type instanceof ConstantStringType) {
-					if (!$allowSameClass && $type->getValue() === $objectOrClassTypeClassName) {
+					if (!$allowSameClass && $objectOrClassTypeClassNames === [$type->getValue()]) {
```

Comment:
array_unique does not preserve keys. I think therefore this comparison might not work, even when equal named classes are compared because keys might be different?

https://3v4l.org/DJ1n6

Reply:
true, thx. actually this is one of the weird cases where it's only relevant if it's one value. I don't quite like my current solution and might come back to this again..